crates/bevy_asset/src/asset_server.rs

```diff
@@ -24,9 +24,9 @@ pub enum AssetServerError {
     #[error("the given type does not match the type of the loaded asset")]
     IncorrectHandleType,
     #[error("encountered an error while loading an asset: {0}")]
-    AssetLoaderError(anyhow::Error),
+    AssetLoaderError(Arc<anyhow::Error>),
     #[error("encountered an error while reading an asset: {0}")]
-    AssetIoError(#[from] AssetIoError),
+    AssetIoError(#[from] Arc<AssetIoError>),
```

Comment:
Hmm ... is `Arc` really the only way to allow this to be `Clone`?

Reply:
not sure..., `Arc` was just the first thing the came to mind.
We could manually go and implement clone for `AssetIoError` but I'm not to sure about that.

Reply:
IMO, `Arc` seems fine. I don't think "checking asset load state" is a particularly hot scenario performance-wise anyway.